hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsenceTypeTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRAbsence_Test_Fabricator_HRAbsenceType as AbsenceTypeFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobLeave as LeaveFabricator;
+use CRM_HRCore_Test_Fabricator_Activity as ActivityFabricator;
+
+/**
+ * Class CRM_HRAbsence_BAO_HRAbsenceTypeTest
+ * Runs tests on AbsenceType BAO Class
+ * 
+ * @group headless
+ */
+class CRM_HRAbsence_BAO_HRAbsenceTypeTest extends PHPUnit_Framework_TestCase 
+  implements HeadlessInterface, TransactionalInterface {
+  
+  protected $absenceType;
+  protected $contact;
+  protected $jobContract;
+  protected $leave;
+  
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('uk.co.compucorp.civicrm.hrcore')
+      ->install('org.civicrm.hrjobcontract')
+      ->installMe(__DIR__)
+      ->apply();
+  }
+    
+  protected function setUp() {
+    // Create Absence Type
+    $this->absenceType = AbsenceTypeFabricator::fabricateUsingAPI();
+    
+    // Create Contact, Contract and Leave for Absence Type
+    $this->contact = ContactFabricator::fabricate(['first_name' => 'john', 'last_name' => 'shepard']);
+    $this->jobContract = HRJobContractFabricator::fabricate(
+      ['contact_id' => $this->contact['id']], 
+      ['period_start_date' => '2015-01-01']
+    );
+    $this->leave = LeaveFabricator::fabricate([
+      'jobcontract_id' => $this->jobContract['id'],
+      'leave_type' => $this->absenceType['id']
+    ]);
+  }
+  
+  /**
+   * Tests that absence typs used in leave requests cannot be deleted.
+   * 
+   * @expectedException CRM_Core_Exception
+   */
+  public function testDeletionOfUsedAbsenceType() {

```

Comment:
For tests, instead of describing what it tests on a docblock, it's better to do it on the test name:

```php
public function testDoesNotAllowAbsenceTypeInUseByLeaveRequestToBeDeleted() {
}
```

If you do that, you can make phpunit output the results in a more readable format, like this:
```
CRM_HRLeaveAndAbsences_Service_EntitlementCalculation
 [x] Brought forward should be zero if absence type doesnt allow carry forward
 [x] Brought forward should be zero if theres no previous period
 [x] Brought forward should be zero if there is no entitlement for previous period
 [x] Brought forward should be zero if expiration duration has expired
 [x] Brought forward should be zero if the previous period is not over yet
 [x] Brought forward should not be more than the max number of days allowed to be carried forward
 [x] Brought forward should not be more than the number of remaining days in previous entitlement
 [x] Pro rata should be zero if the contact has no contracts
 [x] Pro rata should be zero if the contract doesnt have start and end dates
 [x] Pro rata should be rounded to the nearest half day
 [x] Pro rata include public holidays between contract dates even if contract says they should be added
 [x] Pro rata should be the sum of the pro rata for each contract during the absence period
 [x] The proposed entitlement for a contact without a contract should be zero
 [x] The proposed entitlement for a contract without start and end dates should be zero
 [x] The proposed entitlement for a period with previously overridden entitlement should not be the the overridden value
```